frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GraphTableDialog.kt

```diff
@@ -0,0 +1,107 @@
+        private val okAction: Consumer<IGitMacheteRepositorySnapshot>?,
+        private val editAction: Runnable?,
+        private val cancelButtonVisible: Boolean,
+        private val windowTitle: String,
+        private val okButtonText: String
+    ) : DialogWrapper(/* canBeParent */ false) {
+
+  init {
+    super.init()
+    title = windowTitle
+    setOKButtonText(okButtonText)
+  }
+
+  companion object {
+    // @JvmOverloads annotation instructs the Kotlin compiler to generate overloads
+    // for this function that substitute default parameter values (dimension in our case).
+    @JvmOverloads
+    fun of(
+        gitMacheteRepositorySnapshot: IGitMacheteRepositorySnapshot,
+        windowTitle: String,
+        dimension: JBDimension = JBDimension(/* width */ 800, /* height */ 500),
+        emptyTableText: String?,
+        okAction: Consumer<IGitMacheteRepositorySnapshot>?,
+        editAction: Runnable?,
+        okButtonText: String,
+        cancelButtonVisible: Boolean
+    ) =
+        RuntimeBinding.instantiateSoleImplementingClass(ISimpleGraphTableProvider::class.java)
+            .deriveInstance(gitMacheteRepositorySnapshot, /* toggleListingCommits */ false)
+            .apply { setTextForEmptyTable(emptyTableText ?: "") }
+            .let {
+              GraphTableDialog(
+                  it,
+                  gitMacheteRepositorySnapshot,
+                  dimension,
+                  okAction,
+                  editAction,
+                  cancelButtonVisible,
+                  windowTitle,
+                  okButtonText)
+            }
+
+    fun ofDemoRepository() =
+        RuntimeBinding.instantiateSoleImplementingClass(ISimpleGraphTableProvider::class.java)
+            .deriveDemoInstance()
+            .let {
+          GraphTableDialog(
+              it,
+              repositorySnapshot = null,
+              dimension = JBDimension(/* width */ 800, /* height */ 250),
+              okAction = null,
+              editAction = null,
+              cancelButtonVisible = false,
+              windowTitle = "Git Machete Help",
+              okButtonText = "Close")
+        }
+  }
+
+  override fun createActions() =
+      if (cancelButtonVisible)
+          arrayOf(getEditAction(), getOKAction(), cancelAction).filterNotNull().toTypedArray()
+      else arrayOf(getOKAction())
+
+  private fun getEditAction() =
+      if (editAction == null) null
+      else
+          object : AbstractAction("Edit") {
+            override fun actionPerformed(e: ActionEvent?) {
+              doOKAction()
+              editAction.run()
+              close(OK_EXIT_CODE)
+            }
+          }
+
+  override fun createCenterPanel() =
+      JBUI.Panels.simplePanel(/* hgap */ 0, /* vgap */ 2).apply {
+        addToCenter(ScrollPaneFactory.createScrollPane(this@GraphTableDialog.table))
+        preferredSize = dimension
+      }
+
+  @Override
+  override fun doOKAction() {
+    if (getOKAction().isEnabled) {
```

Comment:
I don't get it, how can `OKAction` be NOT enabled when `doOKAction` is called? :thinking:

Reply:
right